fastplotlib/layouts/_gridplot.py

```diff
@@ -92,12 +92,35 @@ def __init__(
                 self.shape
             )
 
+        if "names" in kwargs.keys():
+            self.names = to_array(kwargs["names"])
+            if self.names.shape != self.shape:
+                raise ValueError(f"subplot names: {self.names} must be in gridplot shape: {self.shape}")
+        else:
+            self.names = None
+
         if isinstance(controllers, str):
             if controllers == "sync":
                 controllers = np.zeros(
                     self.shape[0] * self.shape[1], dtype=int
                 ).reshape(self.shape)
 
+        # Check if 'names' and 'controllers' both have values; check if values in controller array are strings
+        if (self.names is not None) and (controllers is not None):
+            c = to_array(controllers)
+            if isinstance(c[0, 0], str):
```

Comment:
do a check to make sure all are str

Reply:
should to_array function set dtype = "o" to avoid typecasting all elements to same type then? Right now it calls np.array(a) with no dtype set, so if even 1 str is in the list, all list elements are converted to str

Reply:
dtype=object *